qcp/matrix.py

```diff
@@ -0,0 +1,150 @@
+# Copyright 2022 Tiernan8r
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import annotations
+from abc import ABC
+from typing import Union
+from ._types import SCALARS, VECTOR, MATRIX
+
+class Matrix(ABC):
+    """
+    Method stubs for an immutable implementation of a matrix.
+    """
+
+    def __init__(self, state: MATRIX):
+        self.state = state
+        self.conjugated = False
```

Comment:
Is this conjugated variable ever used?

Reply:
Not sure, might need @JabethM opinion

Reply:
If it's not needed you can remove it, I just initially thought it might be helpful.

Reply:
The only place I see it reference is here https://github.com/Tiernan8r/quantum_computing_project/pull/25/files#diff-bf5ed9ed646bcabe407cb419b345dc3b1c87058e6b63b8f338f7bd210276a88bR126, but it is never used for any calculations.